keps/sig-network/2732-np-version/README.md

```diff
@@ -0,0 +1,803 @@
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Because NetworkPolicy is implemented by components outside of the
+Kubernetes core, the existing versioning mechanisms in Kubernetes do
+not work well for it; instead of having a two-way version skew between
+"what features the user wants to use" and "what features the cluster
+supports", there is a three-way split between "what features the user
+wants to use", "what features the apiserver knows about", and "what
+features the NetworkPolicy implementation supports". Especially, there
+is often a lag between when new NetworkPolicy features are defined and
+when they are widely implemented. This means that in some cases a user
+cannot know if a cluster will correctly process a given NetworkPolicy
+(and makes it difficult to add new NetworkPolicy features if we want
+to ensure they won't be catastrophically misinterpreted by older
+implementations).
+
+```
+<<[UNRESOLVED snowflake ]>>
+
+Just how special/unusual is NetworkPolicy in this regard? Have any
+other SIGs already dealt with similar problems? Or if not are there
+any other APIs that are similarly failing to deal with the same
+problems?
+
+<<[/UNRESOLVED]>>
+```
+
+Somewhat related to this, there is currently no way for a
```

Comment:
Overlap with https://github.com/kubernetes/enhancements/pull/2947 ?

Should these be split or merged?